cpp/modmesh/math/Complex.hpp

```diff
@@ -35,52 +35,57 @@ namespace modmesh
 
 namespace detail
 {
-template <typename T, typename = std::enable_if_t<std::is_arithmetic<T>::value>>
+
+template <typename T>
 struct ComplexImpl
 {
+    static_assert(std::is_floating_point_v<T>);
+
     T real_v;
     T imag_v;
 
-    ComplexImpl<T> operator+(const ComplexImpl<T> & other) const
+    ComplexImpl & operator+=(const ComplexImpl & other)
     {
-        ComplexImpl<T> ret(*this);
-        return ret += other;
+        real_v += other.real_v;
+        imag_v += other.imag_v;
+        return *this;
     }
 
-    ComplexImpl<T> operator-(const ComplexImpl<T> & other) const
+    ComplexImpl & operator+=(T other)
     {
-        ComplexImpl<T> ret(*this);
-        return ret -= other;
+        real_v += other;
+        return *this;
     }
 
-    ComplexImpl<T> operator*(const ComplexImpl<T> & other) const
+    ComplexImpl & operator-=(const ComplexImpl & other)
     {
-        ComplexImpl<T> ret(*this);
-        ret *= other;
-        return ret;
+        real_v -= other.real_v;
+        imag_v -= other.imag_v;
+        return *this;
     }
 
-    ComplexImpl<T> operator/(const ComplexImpl<T> & other) const
+    ComplexImpl & operator-=(T other)
     {
-        ComplexImpl<T> ret(*this);
-        return ret /= other;
+        real_v -= other;
+        return *this;
     }
 
-    ComplexImpl<T> operator/(const T & other) const
+    ComplexImpl & operator*=(const ComplexImpl<T> & rhs)
     {
-        ComplexImpl<T> ret(*this);
-        return ret /= other;
+        T real_v_copy = real_v;
+        real_v = real_v_copy * rhs.real_v - imag_v * rhs.imag_v;
```

Comment:
Change from 

```cpp
real_v = real_v * rhs.real_v - imag_v * rhs.imag_v;
imag_v = real_v_copy * rhs.imag_v + imag_v * rhs.real_v;
```

to 

```cpp
real_v = real_v_copy * rhs.real_v - imag_v * rhs.imag_v;
imag_v = real_v_copy * rhs.imag_v + imag_v * rhs.real_v;
```

to make the two lines align better for ease of reading.